kinto/plugins/accounts/authentication.py

```diff
@@ -34,13 +41,62 @@ def account_check(username, password, request):
     except storage_exceptions.ObjectNotFoundError:
         return None
 
+    if validation_enabled and not is_validated(existing):
+        return None
+
     hashed = existing["password"].encode(encoding="utf-8")
     pwd_str = password.encode(encoding="utf-8")
     # Check if password is valid (it is a very expensive computation)
     if bcrypt.checkpw(pwd_str, hashed):
         cache.set(cache_key, hashed_password, ttl=cache_ttl)
         return True
 
+    # Last chance, is this a "reset password" flow?
+    reset_password_flow(username, password, request)
```

Comment:
Why no return value here? Deserves a comment IMO

Reply:
I believe I forgot to add a `return` value here, good catch ;)